cli/src/config_manager.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Updating node configuration on the fly.
+
+use exonum::{helpers::config::ConfigManager, node::ConnectListConfig};
+use failure;
+use log::error;
+
+use std::{path::Path, sync::mpsc, thread};
+
+use crate::{
+    config::NodeConfig,
+    io::{load_config_file, save_config_file},
+};
+
+/// Structure that handles work with config file at runtime.
+#[derive(Debug)]
+pub struct DefaultConfigManager {
+    handle: thread::JoinHandle<()>,
+    tx: mpsc::Sender<UpdateRequest>,
+}
+
+/// Messages for ConfigManager.
+#[derive(Debug)]
+pub struct UpdateRequest(ConnectListConfig);
+
+impl DefaultConfigManager {
+    /// Creates a new `ConfigManager` instance for the given path.
+    pub fn new<P>(path: P) -> Self
+    where
+        P: AsRef<Path> + Send + 'static,
+    {
+        let (tx, rx) = mpsc::channel();
+        let handle = thread::spawn(move || {
```

Comment:
Does `handle` do anything? In the old implementation, it was `join`ed on stop, but now this code is seemingly gone. Unlike some resources, threads are not "released" if the handle is dropped.